### PR TITLE
[feat][feat] delta neutral vault deposit test case

### PR DESCRIPTION
**Pull request by sora-alpaca**

### CODE REVIEW

File: test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts [hunk 1]
```diff
@@ -833,6 +837,617 @@ describe("DeltaNeutralVault", () => {
           const aliceShare = await deltaVault.balanceOf(aliceAddress);
           console.log("aliceShare", aliceShare);
         });
+
+        context("when received shares greater than minimum shares should user receive", async () => {
```
Comment (alpanaca) [inline test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts:841]: Why if received share > minimum share expect it should reverted? Should be < minimum

File: test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts [hunk 2]
```diff
@@ -833,6 +837,617 @@ describe("DeltaNeutralVault", () => {
...
+                deltaVaultAsAlice.deposit(zeroAddress(), 0, stableTokenAmount, assetTokenAmount, data, {
+                  value: assetTokenAmount,
+                })
+              ).to.be.revertedWith("ERC20: mint to the zero address");
+            });
+          });
+        });
+
+        describe("_doWork", async () => {
+          context("alice try open position with another position id", async () => {
```
Comment (alpanaca) [inline test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts:987]:
```suggestion
          context("alice try open position with another different position id", async () => {
```

File: test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts [hunk 3]
```diff
@@ -833,6 +837,617 @@ describe("DeltaNeutralVault", () => {
...
+                deltaVaultAsAlice.deposit(zeroAddress(), 0, stableTokenAmount, assetTokenAmount, data, {
+                  value: assetTokenAmount,
+                })
+              ).to.be.revertedWith("ERC20: mint to the zero address");
+            });
+          });
+        });
+
+        describe("_depositHealthCheck", async () => {
+          context("when alice deposit with unsafe position equity on stable side", async () => {
```
Comment (alpanaca) [inline test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts:1057]:
```suggestion
          context("when alice deposit with actions that resulted in unsafe position equity on stable side", async () => {
```

File: test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts [hunk 4]
```diff
@@ -833,6 +837,617 @@ describe("DeltaNeutralVault", () => {
...
+              });
+              await expect(
+                deltaVaultAsAlice.deposit(aliceAddress, 0, stableTokenAmount, assetTokenAmount, data, {
+                  value: assetTokenAmount,
+                })
+              ).to.be.revertedWith("UnsafePositionEquity()");
+            });
+          });
+
+          context("when alice deposit with unsafe position equity on asset side", async () => {
```
Comment (alpanaca) [inline test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts:1117]:
```suggestion
          context("when alice deposit with actions that resulted in unsafe position equity on asset side", async () => {
```

File: test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts [hunk 5]
```diff
@@ -833,6 +837,617 @@ describe("DeltaNeutralVault", () => {
...
+              });
+              await expect(
+                deltaVaultAsAlice.deposit(aliceAddress, 0, stableTokenAmount, assetTokenAmount, data, {
+                  value: assetTokenAmount,
+                })
+              ).to.be.revertedWith("UnsafePositionEquity()");
+            });
+          });
+
+          context("when alice deposit with unsafe debt value on stable side", async () => {
```
Comment (alpanaca) [inline test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts:1177]:
```suggestion
          context("when alice deposit with actions that resulted in unsafe debt value on stable side", async () => {
```

File: test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts [hunk 6]
```diff
@@ -833,6 +837,617 @@ describe("DeltaNeutralVault", () => {
...
+              });
+              await expect(
+                deltaVaultAsAlice.deposit(aliceAddress, 0, stableTokenAmount, assetTokenAmount, data, {
+                  value: assetTokenAmount,
+                })
+              ).to.be.revertedWith("UnsafeDebtValue()");
+            });
+          });
+
+          context("when alice deposit with unsafe debt value on asset side", async () => {
```
Comment (alpanaca) [inline test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts:1237]:
```suggestion
          context("when alice deposit with actions that resulted unsafe debt value on asset side", async () => {
```

File: test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts [hunk 7]
```diff
@@ -833,6 +837,617 @@ describe("DeltaNeutralVault", () => {
...
+                deltaVaultAsAlice.deposit(aliceAddress, 0, stableTokenAmount, assetTokenAmount, data, {
+                  value: assetTokenAmount,
+                })
+              ).to.be.revertedWith("UnsafeDebtValue()");
+            });
+          });
+        });
+
+        describe("_outstandingCheck", () => {
+          context("when stable token amount has descresed", async () => {
```
Comment (alpanaca) [inline test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts:1299]:
```suggestion
          context("when stable token amount in contract has descresed after deposit", async () => {
```

File: test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts [hunk 8]
```diff
@@ -833,6 +837,617 @@ describe("DeltaNeutralVault", () => {
...
+              await expect(
+                deltaVaultAsAlice.deposit(aliceAddress, 0, stableTokenAmount, assetTokenAmount, data, {
+                  value: assetTokenAmount,
+                })
+              ).to.be.revertedWith(
+                `UnsafeOutstanding("${baseToken.address}", ${beforeBaseTokenAmount.toString()}, ${expected.toString()})`
+              );
+            });
+
+            context("when asset token amount has descresed", async () => {
```
Comment (alpanaca) [inline test/integrations/protocol/delta-neutral/DeltaNeutralVault.test.ts:1374]:
```suggestion
            context("when asset token amount in the contract has descresed after deposit", async () => {
```

**Review by alpanaca — APPROVED** [review]

**Review by jr-alpaca — APPROVED** [review]

**Pull request merged**